src/requests/features.jl

```diff
@@ -183,51 +183,25 @@ function format_text(text::AbstractString, params, config)
     end
 end
 
+function mark_range(text::AbstractString, startline, stopline, startmark, stopmark)
+    textlines = split(text, "\n")
+    insert!(textlines, startline + 1, startmark)
+    insert!(textlines, stopline + 3, stopmark)
+    return join(textlines, "\n")
+end
+
 function textDocument_range_formatting_request(params::DocumentRangeFormattingParams, server::LanguageServerInstance, conn)
     doc = getdocument(server, params.textDocument.uri)
-    cst = getcst(doc)
-
-    expr = get_inner_expr(cst, get_offset(doc, params.range.start):get_offset(doc, params.range.stop))
-
-    if expr === nothing
-        return nothing
-    end
-
-    while !(expr.head in (:for, :if, :function, :module, :file, :call) || CSTParser.isassignment(expr))
-        if expr.parent !== nothing
-            expr = expr.parent
-        else
-            return nothing
-        end
-    end
-
-    _, offset = get_file_loc(expr)
-    l1, c1 = get_position_from_offset(doc, offset)
-    c1 = 0
-    start_offset = index_at(doc, Position(l1, c1))
-    l2, c2 = get_position_from_offset(doc, offset + expr.span)
-
-    fulltext = get_text(doc)
-    text = fulltext[start_offset:prevind(fulltext, start_offset+expr.span)]
-
-    longest_prefix = nothing
-    for line in eachline(IOBuffer(text))
-        (isempty(line) || occursin(r"^\s*$", line)) && continue
-        idx = 0
-        for c in line
-            if c == ' ' || c == '\t'
-                idx += 1
-            else
-                break
-            end
-        end
-        line = line[1:idx]
-        longest_prefix = CSTParser.longest_common_prefix(something(longest_prefix, line), line)
-    end
-
-    newcontent = try
+    oldcontent = get_text(doc)
+    startline = params.range.start.line
+    stopline = params.range.stop.line
+    startmark = "#___________START___________"
+    stopmark = "#___________STOP____________"
```

Comment:
Not sure how relevant it is, but we could use UUIDs here to reduce the chance of a collision with genuine comments. Or scan the text for these first I guess.

Reply:
Perhaps just use something more specific like
```suggestion
    startmark = "#___________LANGUAGESERVER_RANGE_FORMATTING_START___________"
    stopmark = "#___________LANGUAGESERVER_RANGE_FORMATTING_STOP____________"
```

Reply:
I ended up using the UUID approach. Let me know what you think.